test/api/model/initial_snapshot_test.dart

```diff
@@ -40,4 +40,24 @@ void main() {
       'unread_message_ids': [11, 2, 3],
     })).throws<AssertionError>();
   });
+
+  test('UnreadMessagesSnapshot from json recognizes channels as streams', () {
```

Comment:
nit: keep test code organized to match the code it's testing, rather than in chronological order

This class is defined above the other three classes tested here, so its test should be above those too.